src/prefect/engine/task_runner.py

```diff
@@ -983,7 +983,13 @@ def get_task_run_state(
             and value is not None
         ):
             try:
-                result = self.result.write(value, filename="output", **prefect.context)
+                result = self.result.write(
+                    value,
+                    filename="output",
+                    **prefect.context.get("parameters", {}),
```

Comment:
I think this can cause a runtime error if you have a task input with the same name as a flow parameter, can it?

Reply:
Good catch! Will fix. The ultimate goal is for the key priority to be context > inputs > parameters.